thirdparty/s3.go

```diff
@@ -43,3 +46,28 @@ func PreSign(r RequestParams) (string, error) {
 	urlStr, err := req.Presign(PresignExpireTime)
 	return urlStr, err
 }
+
+func GetHeadObject(r RequestParams) (*s3.HeadObjectOutput, error) {
```

Comment:
Is it worth moving this functionality into Pail? The repo exists to abstract away S3 operations anyways so that we don't have as much miscellaneous third-party code.

Reply:
What do you mean? We wrote [pail](https://github.com/evergreen-ci/pail), so we can add whatever S3-related functionality we want.

Reply:
My bad, I didn't realize that. In that case, we should probably also move over PreSign. I think it makes sense to create a ticket for it.

Reply:
https://jira.mongodb.org/browse/EVG-16477